aten/src/ATen/Context.h

```diff
@@ -1,13 +1,18 @@
 #pragma once
 
-#include <memory>
-#include <mutex>
 #include "ATen/ATenGeneral.h"
 #include "ATen/Generator.h"
 #include "ATen/Type.h"
 #include "ATen/Utils.h"
 
+#include <memory>
+#include <mutex>
+#include <cstdint>
+
 struct THCState;
```

Comment:
Here would be a pretty good place for a comment saying why we manually pre-declare these types, rather than include cuda.h